pkg/operator/target_config_reconciler.go

```diff
@@ -0,0 +1,279 @@
+		APIVersion: "operator.openshift.io/v1",
+		Kind:       "JobSetOperator",
+		Name:       jobSetOperator.Name,
+		UID:        jobSetOperator.UID,
+	}
+
+	specAnnotations := map[string]string{
+		"jobsetoperators.operator.openshift.io/cluster": strconv.FormatInt(jobSetOperator.Generation, 10),
+	}
+
+	_, _, err = t.manageWebhookService(ctx, jobSetOperator, ownerReference)
+	if err != nil {
+		return err
+	}
+	webhookSecret, _, err := t.manageWebhookCertSecret(ctx, jobSetOperator)
+	if err != nil {
+		return err
+	}
+	specAnnotations["secrets/"+webhookSecret.Name] = webhookSecret.ResourceVersion
+
+	_, _, err = t.manageValidatingWebhook(ctx, jobSetOperator, ownerReference)
+	if err != nil {
+		return err
+	}
+	_, _, err = t.manageMutatingWebhook(ctx, jobSetOperator, ownerReference)
+	if err != nil {
+		return err
+	}
+	_, _, err = t.manageCRD(ctx, jobSetOperator, ownerReference)
+	if err != nil {
+		return err
+	}
+	configMap, _, err := t.manageConfigMap(ctx, jobSetOperator, ownerReference)
+	if err != nil {
+		return err
+	}
+	specAnnotations["configmaps/"+configMap.Name] = configMap.ResourceVersion
+
+	deployment, _, err := t.manageDeployment(ctx, jobSetOperator, ownerReference, specAnnotations, configMap)
+	if err != nil {
+		return err
+	}
+
+	_, _, err = v1helpers.UpdateStatus(ctx, t.jobSetOperatorClient, func(status *operatorv1.OperatorStatus) error {
+		resourcemerge.SetDeploymentGeneration(&status.Generations, deployment)
+		return nil
+	})
+	return err
+}
+
+func (t *TargetConfigReconciler) manageCRD(ctx context.Context, jobSetOperator *jobsetoperatorsv1.JobSetOperator, ownerReference metav1.OwnerReference) (*apiextensionsv1.CustomResourceDefinition, bool, error) {
+	crd := resourceread.ReadCustomResourceDefinitionV1OrDie(bindata.MustAsset("assets/jobset-controller-generated/apiextensions.k8s.io_v1_customresourcedefinition_jobsets.jobset.x-k8s.io.yaml"))
+	if crd.Spec.Conversion != nil && crd.Spec.Conversion.Webhook != nil && crd.Spec.Conversion.Webhook.ClientConfig != nil && crd.Spec.Conversion.Webhook.ClientConfig.Service != nil {
+		crd.Spec.Conversion.Webhook.ClientConfig.Service.Namespace = jobSetOperator.Namespace
+	}
+	crd.OwnerReferences = []metav1.OwnerReference{ownerReference}
+	if crd.Annotations == nil {
+		crd.Annotations = map[string]string{}
+	}
+	crd.Annotations["service.beta.openshift.io/inject-cabundle"] = "true"
+
+	currentCRD, err := t.apiextensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, crd.Name, metav1.GetOptions{})
+	switch {
+	case errors.IsNotFound(err):
+		// no action needed
+	case err != nil && !errors.IsNotFound(err):
+		return nil, false, err
+	case err == nil:
+		if crd.Spec.Conversion != nil && crd.Spec.Conversion.Webhook != nil && crd.Spec.Conversion.Webhook.ClientConfig != nil {
+			crd.Spec.Conversion.Webhook.ClientConfig.CABundle = currentCRD.Spec.Conversion.Webhook.ClientConfig.CABundle
+		}
+	}
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
This is handled in line:153. No need to check err again?.

Reply:
Oh, yes. Nice find!